pkg/apis/components/v1alpha1/types.go

```diff
@@ -60,3 +63,19 @@ type ComponentList struct {
 
 	Items []Component `json:"items"`
 }
+
+// DynamicValue is a dynamic value struct for the component.metadata pair value
+type DynamicValue struct {
+	v1beta1.JSON
```

Comment:
so now user can pass any types of json-object as value, right?

Reply:
Are there any specific validations that must be performed in for the value field?

Reply:
> so now user can pass any types of json-object as value, right?

Yep.

Reply:
> Are there any specific validations that must be performed in for the value field?

The Kubernetes Go SDK requires that strings are saved as JSON strings when creating a new `Component` object programmatically. We only do that in our e2e test setup.